### PR TITLE
Tests for book and category

### PR DESCRIPTION
**Pull request by Den-moroz**

### CODE REVIEW

**Review by Den-moroz — COMMENTED** [review]
image
image
image
image

File: src/main/java/core/basesyntax/bookstore/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -35,21 +39,36 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
         return new ResponseEntity<>(errorResponse, headers, status);
     }
 
+    @ExceptionHandler(SQLIntegrityConstraintViolationException.class)
+    public ResponseEntity<Object> handleSqlIntegrityConstraintViolationException(
+            SQLIntegrityConstraintViolationException ex
+    ) {
+        ErrorResponse errorResponse = new ErrorResponse(LocalDateTime.now(), HttpStatus.NOT_FOUND,
+                List.of(ex.getMessage()));
+        return new ResponseEntity<>(errorResponse, HttpStatus.NOT_FOUND);
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/bookstore/exception/CustomGlobalExceptionHandler.java]: I think it should be other response status (409 for example)

File: src/test/java/core/basesyntax/bookstore/book/BookRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
```
Comment (vovasalyha) [inline src/test/java/core/basesyntax/bookstore/book/BookRepositoryTest.java:57]: Connection is by default autoCommit if I'm not mistaken

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**